resources/acsl/datastructures.md

```diff
@@ -40,32 +56,169 @@
+| 3. <img src="https://user-images.githubusercontent.com/60682642/88591135-7eea3d80-d021-11ea-9dbf-8dcbefdef609.png" width="150" height="150" /> | *E* is added into the tree. No shifting of nodes occurs since *E* is indeed larger than *A*. | 7. <img src="https://user-images.githubusercontent.com/60682642/88591782-7b0aeb00-d022-11ea-9fa9-c1d86b5e9162.png" width="150" height="150" /> | *A* (not the root) is added as a child of *C*. Since it is less than *C*, the two switch places. Although *A* has the same value as the root *A*, switching the two wouldn't make any difference. So the two *A*s can be kept as is. |
+| 4. <img src="https://user-images.githubusercontent.com/60682642/88591248-aa6d2800-d021-11ea-8654-c263398b894c.png" width="150" height="150" /> | *R* is added into the tree and starts the next row since the previous row was completely filled. No shifts occur. | 8. <img src="https://user-images.githubusercontent.com/60682642/88591897-b60d1e80-d022-11ea-923e-6dde512cebd9.png" width="150" height="150" /> | *N* is added as a child of *R* and starts the next row. Since it is less than *R*, the two switch places. *N* is greater than *I*, so it is does move further up the heap. We have now finishing constructing the tree. |
+
 ## Deleting Nodes
 
-<br>
+Just as a reminder, only the root can be deleted! To do so, replace the root with the rightmost node in the last node. Then, keep switching out the root with its smaller child (or larger child, if using max-heap) until the tree is finally valid. Make sure to make adjustions to the rest of the tree if needed. 
 
-# Path Lengths
+In the case that both of the root's children are smaller than the root itself, then choose the smallest child to replace the root with; of course, for max-heap, it would be that if both children are larger than the root, then the larger child should replace the root.
 
 <br>
 
 # Sample Problems
 
 <br>
+
+## 1. Stacks/Queues: Determine what the value of C would be if the operations below were performed on an initially empty stack. Analyze it as a queue afterward.
+
+```code
+PUSH(18)
+B = POP()
+PUSH(7)
+PUSH(33)
+C = POP()
+A = POP()
+```
+
+As a stack... 18 would first be added to the stack; this would then be popped out and stored in B. Afterwards, 7 and 33 (in that order) are put into the stack. Because 33 was pushed into the stack later than 7, it is also the first to be popped; so, 33 is popped and stored into C, whereas 7 is stored in A. So, our final answer is **33**.
+
+As a queue... 18 would be added first; then, this would be popped out and stored in B. Afterwards, 7 and 33 (in that order) are put into the queue. Since queues follow a "first in, first out" rule, 7 would be popped first and stored into C. 33 would consequently be stored in A. So, our final answer here would be **7**.
+
+## 2. Create a priority queue of the letters COMPUTING with a min-heap. What are the letters in the bottommost row, from left to right?
```

Comment:
Just a note: Don't include any questions without solutions. Those will go into the practice folder.

Reply:
Every question that's explained belongs in resources; we'll release unanswered questions on the Practice page, then release answers afterwards.

Reply:
I thought I did explain it though???